src/include/function/gds/gds_frontier.h

```diff
@@ -319,6 +393,74 @@ class KUZU_API DoublePathLengthsFrontierPair : public FrontierPair {
     void beginNewIterationInternalNoLock() override;
 };
 
+class KUZU_API KCoreFrontierPair : public FrontierPair {
+public:
+    KCoreFrontierPair(common::table_id_map_t<common::offset_t> numNodesMap, uint64_t totalNumNodes, uint64_t maxThreadsForExec, storage::MemoryManager* mm);
+
+    void initRJFromSource(common::nodeID_t source) override {};
+
+    void beginFrontierComputeBetweenTables(common::table_id_t curTableID, common::table_id_t nextTableID) override;
+
+    void beginNewIterationInternalNoLock() override;  
+
+    uint64_t addToVertexDegree(common::nodeID_t nodeID, uint64_t degreeToAdd) {
+        return getVertexDegreeAtomic(nodeID).fetch_add(degreeToAdd, std::memory_order_relaxed);
+    }
+
+    uint64_t removeFromVertex(common::nodeID_t nodeID) {
+        int curSmallest = curSmallestDegree.load(std::memory_order_relaxed);
+        int curVertexDegree = getVertexDegree(nodeID);
+        if (curVertexDegree > curSmallest) {
+            return getVertexDegreeAtomic(nodeID).fetch_sub(1, std::memory_order_relaxed);
+        }
+        return curVertexDegree;
+    }
+
+    uint64_t getVertexDegree(common::nodeID_t nodeID) const {
+        return getVertexDegreeAtomic(nodeID).load(std::memory_order_relaxed);
+    }
+
+    bool isNodeActive(common::nodeID_t nodeID) const {
```

Comment:
Why not use/overwrite isActive/setActive but instead create isNodeActive/setNodeActive?